File: app/src/main/java/be/vergauwen/simon/cleanmvp/ui/TestContract.java
package be.vergauwen.simon.cleanmvp.ui;

import be.vergauwen.simon.cleanmvp.mvp.MVPContract;

interface TestContract {
    interface View extends MVPContract.View {
        void showThings();
    }

    interface Presenter extends MVPContract.Presenter<View> {
        void loadThings();
    }
}

File: app/src/main/kotlin/be/vergauwen/simon/cleanmvp/ui/TestContract.kt
package be.vergauwen.simon.cleanmvp.ui

import be.vergauwen.simon.cleanmvp.mvp.MVPContract

interface TestContract{
  interface View : MVPContract.View{
    fun showThings()
  }

  interface Presenter : MVPContract.Presenter<View>{
    fun loadThings()
  }
}